ansible/roles/nginx/templates/nginx.conf.j2

```diff
@@ -12,10 +12,16 @@ http {
 {# allow large uploads, need to thread proper limit into here #}
     client_max_body_size 50M;
 
+    map $http_ow_tid $ow_tid {
```

Comment:
Can we use [set_secure_random_alphanum](https://github.com/openresty/set-misc-nginx-module#set_secure_random_alphanum) instead ?

Reply:
And if we want to respect an ID sent by the client, then we can use a config similar to : https://github.com/adobe-apiplatform/apigateway/blob/master/api-gateway-config/conf.d/commons/common-headers.conf#L23-L30 which only sets an ID unless one is provided. or is there any reason to have this ID adhere to a strict format, instead of having it treated as a String ?

Reply:
@ddragosd `set_secure_random_alphanum` depends on `/dev/urandom` which may impact performance. Is there a requirement to use a secure random and not just the ngninx built-in `request_id`? We should also understand the performance of `set_secure_random_alphanum` compared to `request_id`.

Reply:
@sven-lange-last 👍   NGINX's `request_id` should do it as well. I've been using the other `set_secure_random_alphanum` in the past years ( before NGINX 1.11.x added `request_id` ) and the performance implications were not visible. 

One thing to consider here is that OW should accept a variable length for this ID ( which IMHO should be fine) b/c NGINX might update this in the future; I'm not aware of a way to configure the size `16` or `32` or another size, unlike with the older `set_secure....`; Note that the article referenced says it's 32 chars, while the [docs](http://nginx.org/en/docs/http/ngx_http_core_module.html) say it's 16.

Reply:
Agreed on the variable length. It shouldn't care at all (so nobody is forced to use this specific config as their front-door). Will be changed!